mitodl/open-discussions
# Added video unpublish support

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2362
Closes #2363
Part of #2367 

#### What's this PR do?
- Makes `offered_by` optional in the configuration
- Adds tracking of channels and playlists (this will enable us to easily generate user lists from this data for #2367 in a subsequent PR)
- Adds support for unpublishing videos (the logic here is that if a video no longer appears in any playlist we're tracking we consider it unpublished and should not appear in search)

#### How should this be manually tested?

- Delete all video records
- Run `./manage.py backpopulate_youtube_data fetch`
- Verify you have 5988 OCW videos and 54 mitx videos on the search page
- In `.env`, set `OPEN_VIDEO_DATA_BRANCH=nl/small_playlist`, restart workers
- Run `./manage.py backpopulate_youtube_data fetch`
- Verify you have 34 OCW videos and 54 mitx videos on the search page
